NGitLab.Tests/MembersClientTests.cs

```diff
@@ -151,4 +151,96 @@ public async Task GetAccessLevelMemberOfGroup()
         var groupUser = context.Client.Members.GetMemberOfGroup(groupId, user.Id.ToString(CultureInfo.InvariantCulture));
         Assert.That((AccessLevel)groupUser.AccessLevel, Is.EqualTo(AccessLevel.Developer));
     }
+
+    [Test]
+    public async Task AsyncMethodsBehaveAsExpected()
+    {
+        // Arrange
+        using var context = await GitLabTestContext.CreateAsync();
+        var client = context.Client;
+        var ownerName = client.Users.Current.Username;
+
+        // Create a user, a top-level group, a group project, and a subgroup...
+        context.CreateNewUser(out var user1);
+        var group1 = context.CreateGroup();
+        var group2 = context.CreateSubgroup(group1.Id);
+        var project = context.CreateProject(group1.Id);
+
+        var user1Name = user1.Username;
+        var user1Id = user1.Id.ToString();
+
+        // Add the user to the top-level group as a Maintainer...
+        await client.Members.AddMemberToGroupAsync(group1, new()
+        {
+            AccessLevel = AccessLevel.Maintainer,
+            UserId = user1Id,
+        });
+
+        // Act
+        // Assert
```

Comment:
I know, not a pretty test. I could split it up, but the setup for the smaller tests would just get cumbersome. Hopefully the comments make it easier to reason about.